nautobot/core/apps/__init__.py

```diff
@@ -134,23 +137,31 @@ def register_menu_items(tab_list):
                         sorted(registry_buttons.items(), key=lambda kv_pair: kv_pair[1]["weight"])
                     )
 
-                    group_perms |= set(perms for perms in item.permissions)
+                    # If any item has "no" permissions required, then the group behaves likewise
+                    if group_perms is None or not item.permissions:
+                        group_perms = None
+                    else:
+                        group_perms |= set(perms for perms in item.permissions)
 
                 # Add sorted items to group registry dict
                 registry_groups[group.name]["items"] = OrderedDict(
                     sorted(registry_groups[group.name]["items"].items(), key=lambda kv_pair: kv_pair[1]["weight"])
                 )
                 # Add collected permissions to group
                 registry_groups[group.name]["permissions"] = group_perms
-                # Add collected permissions to tab
-                tab_perms |= group_perms
+
+                # If any group has "no" permissions required, then the tab performs likewise
+                if tab_perms is None or not group_perms:
+                    tab_perms = None
+                else:
+                    tab_perms |= group_perms
 
             # Add sorted groups to tab dict
             registry["nav_menu"]["tabs"][nav_tab.name]["groups"] = OrderedDict(
                 sorted(registry_groups.items(), key=lambda kv_pair: kv_pair[1]["weight"])
             )
             # Add collected permissions to tab dict
-            registry["nav_menu"]["tabs"][nav_tab.name]["permissions"] |= tab_perms
+            registry["nav_menu"]["tabs"][nav_tab.name]["permissions"] = tab_perms
```

Comment:
Does this change prevent apps from being able to add permissions to an existing top level tab?

Reply:
No, the permissions bubble up correctly because of this in combination with the change above at line 101/104 , for example the `example_app` adds a permission to the Circuits tab for its custom nav entry there:

```
>>> from nautobot.extras.registry import registry
>>> registry["nav_menu"]["tabs"]["Circuits"]["permissions"]
{'circuits.view_providernetwork', 'circuits.view_circuit', 'circuits.view_provider', 'example_app.view_examplemodel', 'circuits.view_circuittype'}
```

Reply:
Or just among core apps:

```
>>> registry["nav_menu"]["tabs"]["Organization"]["permissions"]
{'dcim.view_rack', 'dcim.view_rackgroup', 'dcim.view_locationtype', 'dcim.view_rackreservation', 'extras.view_role', 'tenancy.view_tenant', 'extras.view_tag', 'extras.view_dynamicgroup', 'extras.view_status', 'extras.view_contact', 'extras.view_team', 'tenancy.view_tenantgroup', 'dcim.view_location'}
```